pkg/types/validation/installconfig.go

```diff
@@ -50,6 +51,24 @@ func ValidateInstallConfig(c *types.InstallConfig) field.ErrorList {
 	if c.TypeMeta.APIVersion == "" {
 		return field.ErrorList{field.Required(field.NewPath("apiVersion"), "install-config version required")}
 	}
+	if c.Proxy != nil {
+		if c.Proxy.NoProxy != "" && strings.Contains(c.Proxy.NoProxy, " ") {
+			return field.ErrorList{field.Invalid(field.NewPath("noProxy"), c.Proxy.NoProxy, fmt.Sprintf("noProxy must not have spaces"))}
+		}
+		if c.Proxy.HTTPProxy != "" {
+			err := validateURI(c.Proxy.HTTPProxy, field.NewPath("httpProxy"), "HTTPProxy", []string{"http"})
+			if err != nil {
+				return err
+			}
```

Comment:
```suggestion
			allErrs = append(allErrs, validateURI(c.Proxy.HTTPProxy, field.NewPath("httpProxy"), "HTTPProxy", []string{"http"})...)
```

Reply:
Was fixing this. When there's no errors I return nil and that actually gets added to the list which was confusing so I added the if clause.